test/e2e/index.test.ts

```diff
@@ -62,330 +62,490 @@ async function attemptGoto(page: playwright.Page, url: string): Promise<boolean>
+    describe('<DataGrid />', () => {
+      it('should select the first column header when pressing tab key', async () => {
+        await renderFixture('DataGrid/KeyboardNavigationFocus');
 
-      const button = page.locator('text=initial focus');
-      expect(await button.evaluate((node) => document.activeElement === node));
+        const button = page.locator('text=initial focus');
+        expect(await button.evaluate((node) => document.activeElement === node));
 
-      await page.keyboard.press('Tab');
-      await waitFor(async () => {
+        await page.keyboard.press('Tab');
         expect(await page.evaluate(() => document.activeElement?.textContent)).to.equal('brand');
-        expect(await page.evaluate(() => document.activeElement?.getAttribute('role'))).to.equal(
-          'columnheader',
-        );
       });
-      await page.keyboard.press('ArrowDown');
-      await waitFor(async () => {
-        expect(await page.evaluate(() => document.activeElement?.textContent)).to.equal('Nike');
-        expect(await page.evaluate(() => document.activeElement?.getAttribute('role'))).to.equal(
-          'cell',
-        );
-      });
-      await page.keyboard.press('ArrowDown');
-      await waitFor(async () => {
-        expect(await page.evaluate(() => document.activeElement?.textContent)).to.equal('Adidas');
-        expect(await page.evaluate(() => document.activeElement?.getAttribute('role'))).to.equal(
-          'cell',
-        );
+
+      it('should implement the roving tabindex pattern', async () => {
+        await renderFixture('DataGrid/KeyboardNavigationFocus');
+
+        const button = page.locator('text=initial focus');
+        expect(await button.evaluate((node) => document.activeElement === node));
+
+        await page.keyboard.press('Tab');
+        await waitFor(async () => {
+          expect(await page.evaluate(() => document.activeElement?.textContent)).to.equal('brand');
+          expect(await page.evaluate(() => document.activeElement?.getAttribute('role'))).to.equal(
+            'columnheader',
+          );
+        });
+        await page.keyboard.press('ArrowDown');
+        await waitFor(async () => {
+          expect(await page.evaluate(() => document.activeElement?.textContent)).to.equal('Nike');
+          expect(await page.evaluate(() => document.activeElement?.getAttribute('role'))).to.equal(
+            'cell',
+          );
+        });
+        await page.keyboard.press('ArrowDown');
+        await waitFor(async () => {
+          expect(await page.evaluate(() => document.activeElement?.textContent)).to.equal('Adidas');
+          expect(await page.evaluate(() => document.activeElement?.getAttribute('role'))).to.equal(
+            'cell',
+          );
+        });
+        // Firefox insists on focusing the pagination wrapper first before the input.
+        // This might need looking into in more depth.
+        // Maybe it's a bug?
```

Comment:
@mui/xgrid Could you take a look at this and clarify if it's not an actual problem? 🤔

Reply:
I learned something new today - turns out Firefox focuses on scrollable elements so you can scroll them with keyboard arrow keys 🙂

https://github.com/mui/mui-x/assets/13808724/b844af19-3a7b-4581-a7d9-95f31d663b88

I'll open a PR to avoid that in the test.
Is there anything else I should take a look at in this PR?

Reply:
Nice! That's an interesting a11y feature.
No, this was the only interesting behavior I didn't understand the root cause of. 🙈